docs/workflow/building/coreclr/README.md

```diff
@@ -1,56 +1,100 @@
+## Introduction
 
-For Linux:
-```
-./build.sh -subset clr
+Here is a brief overview on how to build the common form of CoreCLR in general. For further specific instructions on each platform, we have links to instructions later on in [Platform-Specific Instructions](#platform-specific-instructions).
+
+To build just CoreCLR, use the `subset` flag to the `build.sh` or `build.cmd` script at the repo root:
+
+For Linux and MacOS:
+
+```bash
+./build.sh --subset clr
 ```
 
 For Windows:
-```
-build.cmd -subset clr
-```
-Specifying `-subset` explicitly is not necessary if it is the first argument: `./build -subset clr` and `./build clr` are equivalent.
 
-By default, build generates a 'debug' build type, that includes asserts and is easier for some people to debug. If you want to make performance measurements, or just want tests to execute more quickly, you can also build the 'release' version (which does not have these checks) by adding the flag `-configuration release` (or `-c release`), for example:
-```
-./build.sh -subset clr -configuration release
+```cmd
+.\build.cmd -subset clr
 ```
 
-CoreCLR also supports a 'checked' build type which has asserts enabled like 'debug', but is built with the native compiler optimizer enabled, so it runs much faster. This is the usual mode used for running tests in the CI system. You can build that using, for example:
+Specifying `-subset` explicitly is not necessary if it is the first argument: `./build.sh --subset clr` and `./build.sh clr` are equivalent. However, if you specify any other argument beforehand, then you must specify the `-subset` flag.
+
+## Common Building Options
+
+By default, the script generates a _Debug_ build type, which is not optimized code and includes asserts. As its name suggests, this makes it easier and friendlier to debug the code. If you want to make performance measurements, you ought to build the _Release_ version instead, which doesn't have any asserts and has all code optimizations enabled. Likewise, if you plan on running tests, the _Release_ configuration is more suitable since it's considerably faster than the _Debug_ one. For this, you add the flag `-configuration release` (or `-c release`). For example:
+
+```bash
+./build.sh --subset clr --configuration release
 ```
-./build.sh -subset clr -configuration checked
+
+As mentioned before in the [general building document](/docs/workflow/README.md#configurations-and-subsets), CoreCLR also supports a _Checked_ build type which has asserts enabled like _Debug_, but is built with the native compiler optimizer enabled, so it runs much faster. This is the usual mode used for running tests in the CI system.
+
+Now, it is also possible to select a different configuration for each subset when building them together. The `--configuration` flag applies universally to all subsets, but it can be overridden with any one or more of the following ones:
```

Comment:
The long-standing problem is that all testing assumes library build is release whereas "build.(sh|cmd)" goes for "debug" by default. It would be nice to somehow explain it here but I don't insist on doing it in this particular PR, you've already gone a long way towards making our build process easier to understand.